sympy/geometry/plane.py

```diff
@@ -418,7 +418,7 @@ def intersection(self, o):
                     p = a.subs(t, c[0])
                     if p not in self:
                         return []  # e.g. a segment might not intersect a plane
-                    return [p]
+                    return [x for x in [p] if x in o]
```

Comment:
I think the condition should be `if p not in o`. This will also need test(s) to be included.

Reply:
I meant this as a logical change. You can investigate further as to what additional changes might be required.

Reply:
I have added the tests. Please have a look.

Reply:
@czgdp1807 wouldn't it be enough to just change the condition in the ``if`` statement two lines above your return statement, such that it reads ``if p not in self or p not in o``? Then the return statement you changed can stay as it was before (``return [p]``) and all the "checking membership" stuff is localized to that one ``if`` statement, not split up in two.

Reply:
The point `p` is obtained using the plane equation [itself](https://github.com/sympy/sympy/blob/master/sympy/geometry/plane.py#L414). So, checking `if p not in self` might not be required.

Reply:
@sidhantnagpal Oh ok, so that's what you meant earlier. I thought you were talking about the list comprehension - maybe @czgdp1807 thought that as well and that's why it didn't work? ``p not in o`` in the ``if`` statement two lines above should in fact work, you're right.

Reply:
@sidhantnagpal I got your point, though late. Thanks for your review. :-)